pydata/pydata-sphinx-theme
# Keyboard handlers to open and close mobile sidebars

**IMPORTANT: this PR is against `feature-focus` branch not `main`.** Style fixes related to interaction state (focus, hover, current) will be collected in small increments into `feature-focus` before being merged into `main`.

Rather than completely refactor the mobile sidebar implementation, I decided to tweak it as little as possible. Here's what I did:

- Change the labels in the nav bar to buttons. 
  - This was necessary because the labels were not focusable and even making them focusable did not make it possible to use the Enter and Space keys. 
- Rewire the buttons to the checkboxes.
  - Changing the labels to buttons severed the link between the label and the checkbox that controls (via CSS) whether the corresponding sidebar is open or closed. So I had to wire up the buttons to the checkboxes. 
- Wire the sidebars up to the Escape key so they can be closed by pressing the Escape key.
  - This also required moving focus to the sidebar when the button is clicked. And returning focus to that button after pressing Escape.
- Listen to the checkboxes, so that when the overlay label is clicked and it changes the checkbox, we can return focus to the appropriate nav bar button.
- Add ARIA labels to the nav bar buttons, reusing existing translated strings.

### Test plan

Go to home page, i.e. a page with only one sidebar and one button to open it.

Reduce the width of the browser until the button appears in the nav bar, then:

- Click on the button, sidebar should slide in
- Press Tab, your focus should already been moved into the sidebar
- Press Escape, the sidebar should close
- Press Tab, you should be taken to the next tab stop after the button that opened the sidebar
- Repeat the steps above, but instead of closing the sidebar with the escape key, click in the area outside the sidebar. The sidebar should close, and focus should be returned to the button that opened the sidebar.

Now go to a page that has both sidebar toggles in the nav bar and repeat the same test steps.
